SharpSource/SharpSource/Diagnostics/AsyncOverloadsAvailableAnalyzer.cs

```diff
@@ -43,8 +43,9 @@ private static void Analyze(OperationBlockAnalysisContext context, INamedTypeSym
             return;
         }
 
-        // Problem: If the surrounding method is a global statement is it considered as `IsAsync: false` even though async calls work
-        if (surroundingMethod is { IsAsync: false })
+        // If the surrounding method is a global statement is it considered as `IsAsync: false` even though async calls work
+        // Using CanBeReferencedByName to try and distinguish between the two scenarios since the top-level statements entry point can't be referenced 
+        if (surroundingMethod is { IsAsync: false, CanBeReferencedByName: true })
```

Comment:
If you want to be very specific about the top-level statement case, you can compare the name against `WellKnownMemberNames.TopLevelStatementsEntryPointMethodName`

Reply:
Excellent, that's what I was hoping for